data-import/desktop-csv-import/desktop-csv-import.adoc

```diff
@@ -0,0 +1,420 @@
+To do that, we use Cypher statements alongside the `LOAD CSV` commands we used above.
+The `LOAD CSV` will read the files, and the Cypher statements will create the data in our database.
+
+=== Graph data model
+
+An important step we need before writing Cypher statements, though, is to determine what the graph structure should look like once we import our file data.
+After all, importing the data in the existing table and column data will not provide the value we want to achieve from a graph.
+To utilize the graph database fully, we need a graph data model.
+
+We do not need to analyze all the various ways we could organize the products and orders in our files, so we have provided the model we will use for this exercise below.
+
+image::{img}desktop_import_data_model.jpg[link="{img}desktop_import_data_model.jpg",role="popup-link"]
+
+We have two nodes - one for a product and one for an order.
+Each of those nodes have properties that we have from our CSV files.
+For the `Product`, we have ID, name, and unit cost.
+For the `Order`, we have ID, date/time, and country where it is going.
+
+The order-details.csv file defines the relationship between those two nodes.
+That file has the product ID, the order ID it belongs to, and the quantity of the product on the order.
+So, in our data model, this becomes our `CONTAINS` relationship between `Product` and `Order` nodes.
+We also include a property of `quantityOrdered` on the relationship because the product quantity value only exists when a product is related to an order.
+
+Now that we know the types of nodes and relationships we will have and the properties involved, we can construct the Cypher statements to create the data for this model.
+
+=== Avoiding duplicates and increasing performance
+
+One final thing we need to think about before we create data in the graph is ensuring values are unique and performance is efficient.
+To handle this, we can use constraints.
+Just as with other databases, constraints ensure data integrity criteria is not violated, while simultaneously indexing the property with the constraint for faster query performance.
+
+In this exercise, we will add two constraints - one for `productId` and one for `orderId`.
+This will ensure when we create a new node of each of those types or a relationship to connect them, we know the entities are unique and indexed.
+
+Good practice is to add any constraints and indexes before loading to avoid existing violations and ensure data is added to the index as it is created.
+Below is the Cypher for adding indexes.
+
+[source, cypher]
+----
+CREATE CONSTRAINT ON (p:Product) ASSERT p.id IS UNIQUE;
+CREATE CONSTRAINT ON (o:Order) ASSERT o.id IS UNIQUE;
+----
+
+=== Cypher
+
+Now we are ready to create the Cypher for creating the data in the graph!
+We will use `CREATE` statements where we are sure that we will not have duplicate rows in our CSV file.
```

Comment:
how are we sure?

eg.. `return count(row.productId), count(distinct row.productId)`